site-src/concepts/security-considerations.md

```diff
@@ -0,0 +1,174 @@
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: Selector
+        selector:
+            kubernetes.io/metadata.name: ns1
+```
+
+* Because the number of listeners is limited on a Gateway, the administrator should
+instead rely on some validation (like ValidationAdmissionPolicy or some other mechanism)
+that limits what hostnames can be used on routes of each namespaces.
+* In case of `ListenerSet` (still experimental) a validation policy should also be applied.
+
+### Example of a ValidatingAdmissionPolicy
+
+A [ValidatingAdmissionPolicy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/)
+can be used to add rules that limits what namespaces can use what domains. 
+
+!!! warning
+    The validation policy shown here **IS AN EXAMPLE** and the cluster-admin should do
+    adjustments to their own environment! Do not copy/paste this example with proper
+    adjustments
+
+The policy exemplified here will:
+
+* Read the allowed domains from a comma-separated value of the `annotation` "domains" present on the namespace.
+* Validate if all of the hostnames within `.spec.hostnames` are contained on this annotation.
+* In case any of the entries are not authorized, the policy denies its admission.
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: httproute-hostname-policy
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups: ["gateway.networking.k8s.io"]
+      apiVersions: ["v1", "v1beta1"]
+      operations: ["CREATE", "UPDATE"]
+      resources: ["httproutes"]
+  variables:
+  - name: allowed_hostnames_str
+    expression: |
+      has(namespaceObject.metadata.annotations) && 
+      has(namespaceObject.metadata.annotations.domains) ?
+      namespaceObject.metadata.annotations['domains'] : '' 
+  - name: allowed_hostnames_list
+    expression: |
+      variables.allowed_hostnames_str.split(',').
+      map(h, h.trim()).filter(h, size(h) > 0)
+  validations:
+  - expression: |
+      !has(object.spec.hostnames) || 
+      size(object.spec.hostnames) == 0 || 
+      object.spec.hostnames.all(hostname, hostname in variables.allowed_hostnames_list)
+    message: "HTTPRoute validation failed. It contains unauthorized hostnames"
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: httproute-hostname-binding
+spec:
+  policyName: httproute-hostname-policy
+  validationActions: ["Deny"]
+  matchResources:
+    namespaceSelector: {}
+```
+
+Once the policy is created, the cluster-admin should explicitly allow the usage of
+domains with a command like `kubectl annotate ns default domains=www.dom1.tld,www.dom2.tld`
+
+Additionally, when dealing with environments that provide DNS record creations,
+admins should be aware and limit the DNS creation based on the same constraints above.
+
+## Usage of ReferenceGrant
```

Comment:
Based on my comments below, it might be worth reframing this to "Limiting Cross-Namespace References"